JoinColony/colonyDapp
# Create `MotionMintTokens` saga

## Description

This PR adds actions and saga for mint tokens motion.

It adds colonyNetwork reputation changes. It also adds script to add proxy for the reputationOracleEndpoint. There is a need for a change in the colony-js but Alex said that we can't do a new release just for that, so he added proxy.

The steps to test:

- `npm i`
- `npm run provision`
- start `dev-heavy` environment
- create a colony
- mint tokens
- make a payment. You need to wait a bit for the reputation to get updated and also reload the web app. It caught me several times. This needs to be investigated more but it's out of the scope of this PR. The motion saga will not work without any reputation in the colony.
- install & enable voting extension
- **mint tokens  - that would trigger `mintTokensMotion`**

**New stuff** ✨

- add mint tokens motion actions
- add mint tokens motion saga
- add condition to mint tokens dialog to check if the voting extension is installed


**Deletions** ⚰️

- none

## TODO

- Prevent mint tokens motion saga to start if there is no reputation in the colony

Resolves DEV-251

![image](https://user-images.githubusercontent.com/34057551/112134596-d21e8b00-8bc4-11eb-96f6-f008375e0346.png)
